ring.go

```diff
@@ -213,6 +213,7 @@ func (shard *ringShard) Vote(up bool) bool {
 type ringSharding struct {
 	opt *RingOptions
 
+	addrsMu   sync.Mutex
```

Comment:
Please try to make it clear what this mutex protects, e.g. I would expect something like

```go
addrsMu   sync.Mutex
addrs        type

// another group
```

Reply:
After reading the code and the PR comment, I suggest to rename this to `setAddrsMu` and add an explanation similar to the one you have in the PR description.